src/app.service.ts

```diff
@@ -12,10 +12,23 @@ export class AppService {
   constructor(private prisma: PrismaService) {}
 
   async getStats(): Promise<any> {
-    return this.prisma.trending.groupBy({
+    const stats = await this.prisma.trending.groupBy({
       by: ['createdAt', 'type'],
       _count: { id: true },
     });
+    if (stats.length > 0) {
+      return stats;
+    }
+    const yesterday = new Date();
+    yesterday.setDate(yesterday.getDate() - 1);
+
+    const year = yesterday.getFullYear();
+    const month = String(yesterday.getMonth() + 1).padStart(2, '0'); // Months are 0-indexed
+    const day = String(yesterday.getDate()).padStart(2, '0');
+
+    const getYesterdaysDate = `${year}-${month}-${day}`;
```

Comment:
I think we can use `format` like we use already in the file...

https://github.com/EddieHubCommunity/github-trending-repos/blob/fb177abd6728740eada3ad30e363ca101ce825d9/src/app.service.ts#L36

Reply:
I have used the format function. I didn't see it. LOL!

Reply:
thanks